python/instrumentation/openinference-instrumentation-openai/src/openinference/instrumentation/openai/_request.py

```diff
@@ -0,0 +1,358 @@
+import logging
+from abc import ABC
+from contextlib import contextmanager
+from types import MappingProxyType
+from typing import (
+    Any,
+    Awaitable,
+    Callable,
+    Dict,
+    Hashable,
+    Iterator,
+    Mapping,
+    Optional,
+    Tuple,
+)
+
+from openinference.instrumentation.openai._extra_attributes_from_request import (
+    _get_extra_attributes_from_request,
+)
+from openinference.instrumentation.openai._extra_attributes_from_response import (
+    _get_extra_attributes_from_response,
+)
+from openinference.instrumentation.openai._response_accumulator import (
+    _ChatCompletionAccumulator,
+    _CompletionAccumulator,
+)
+from openinference.instrumentation.openai._stream import (
+    _ResponseAccumulator,
+    _Stream,
+)
+from openinference.instrumentation.openai._utils import (
+    _as_input_attributes,
+    _as_output_attributes,
+    _finish_tracing,
+    _io_value_and_type,
+)
+from openinference.instrumentation.openai._with_span import _WithSpan
+from openinference.semconv.trace import SpanAttributes
+from opentelemetry import context as context_api
+from opentelemetry import trace as trace_api
+from opentelemetry.context import _SUPPRESS_INSTRUMENTATION_KEY
+from opentelemetry.trace import INVALID_SPAN
+from opentelemetry.util.types import AttributeValue
+
+from openai import AsyncStream, Stream
+from openai.types import Completion, CreateEmbeddingResponse
+from openai.types.chat import ChatCompletion
+
+__all__ = (
+    "_Request",
+    "_AsyncRequest",
+)
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
```

Comment:
This is to silence it (from stdout/stderr) with a no-op handler (otherwise if it has no handler, by default everything goes back out to stdout/stderr). The user brings it back by reverting it, which you can see my do in the integration tests.